### PR TITLE
error handling

### PR DESCRIPTION
**Pull request by RobertGlein**
Surprisingly easy fix when you know it :)

### CODE REVIEW

File: entrypoint.sh [hunk 1]
```diff
@@ -1,6 +1,7 @@
 #!/bin/sh
 
-set -u
+# Error handling
+set -euo pipefail
```
Comment (aperloff) [inline entrypoint.sh:4]: Please explain what each of these options do as well as the 'pipefail' statement.

File: entrypoint.sh [hunk 2]
```diff
@@ -15,12 +16,7 @@ sh -c "git config --global core.askPass /cred-helper.sh"
 sh -c "git config --global credential.helper cache"
 sh -c "git remote add mirror $*"
 sh -c "echo pushing to $branch branch at $(git remote get-url --push mirror)"
-sh -c "git push mirror $branch"
-
-if [[ -z "${GITLAB_PASSWORD}" ]]; then
-  echo "Empty or non-existent GitLab password."
-  exit 1
-fi
+git push mirror $branch
```
Comment (aperloff) [inline entrypoint.sh]: Why not run this command in its own shell?

File: entrypoint.sh [hunk 3]
```diff
@@ -15,12 +16,7 @@ sh -c "git config --global core.askPass /cred-helper.sh"
 sh -c "git config --global credential.helper cache"
 sh -c "git remote add mirror $*"
 sh -c "echo pushing to $branch branch at $(git remote get-url --push mirror)"
-sh -c "git push mirror $branch"
-
```
Comment (aperloff) [inline entrypoint.sh:19]: Why are you removing this block? Isn't this what you added in your last PR?

Comment (RobertGlein) [reply entrypoint.sh:19]: This is now covered by "set -euo pipefail"

File: entrypoint.sh (hunk 1 above)
Comment (RobertGlein) [reply entrypoint.sh:4]: done

File: entrypoint.sh (hunk 2 above)
Comment (RobertGlein) [reply entrypoint.sh]: I wanted to see if this masks the error and forgot to change it back

**Comment by aperloff** [comment]
And this is fully tested?

**Comment by RobertGlein** [comment]
It did tests with diverging branches, no PW, and regular actions that pass.

**Pull request merged**